### PR TITLE
Add endpoints related to images

### PR DESCRIPTION
**Pull request by ryancarlson**
- list images for a package
 - retrieve a specific image
 - upload a new image

### CODE REVIEW

**Comment by ryancarlson** [comment]
@LaelWatkins can you review this?  I pushed it to stage and I don't see any issues w/ the existing functionality.  I've tested these specific endpoints locally too.

**Comment by LaelWatkins** [comment]
looking at it now

**Comment by ryancarlson** [comment]
One thing i noticed is that the filename didn't get saved.

File: src/main/java/org/cru/godtools/domain/images/ImageService.java [hunk 1]
```diff
@@ -20,18 +18,31 @@ public ImageService(Connection sqlConnection)
...
-        return sqlConnection.createQuery(ImageQueries.selectById)
-                .setAutoDeriveColumnNames(true)
-                .addParameter("id", id)
-                .executeAndFetchFirst(Image.class);
+		Image image = sqlConnection.createQuery(ImageQueries.selectById)
+				.setAutoDeriveColumnNames(true)
+				.addParameter("id", id)
+				.executeAndFetchFirst(Image.class);
+
+		if(image != null && image.getImageContent() != null)
```
Comment (LaelWatkins) [inline src/main/java/org/cru/godtools/domain/images/ImageService.java]: Just a nit pick: what do you think about having a method that does this ``if``? It the same logic on line 43.

File: src/main/java/org/cru/godtools/api/v2/ImageResource.java [hunk 2]
```diff
@@ -0,0 +1,213 @@
...
+
+		return Response
+				.created(URI.create(String.format("/packages/%s/images/%s", packageCode, image.getId())))
+				.build();
+	}
+
+	/**
+	 * Thanks to tutorial by MK Yong: http://www.mkyong.com/webservices/jax-rs/file-upload-example-in-resteasy/
+	 */
+	private String getFileName(MultivaluedMap<String, String> header)
```
Comment (LaelWatkins) [inline src/main/java/org/cru/godtools/api/v2/ImageResource.java]: Just curious, is this method is for future use?

**Comment by LaelWatkins** [comment]
The filename is written to the ``Image`` object by the client? I'm asking because of your comment yesterday. And, it doesn't appear to be added to the object anywhere in the ``ImageResource`` class.

**Comment by LaelWatkins** [comment]
I only have a couple of comments.  This looks great!  There is the question about the filename being set on the ``Image`` object.  Unless I'm mistaken, that doesn't appear to be an issue here in the api.  

If it does require a change, i'll take a look whenever it's ready.  Let me know.

File: src/main/java/org/cru/godtools/domain/images/ImageService.java (hunk 1 above)
Comment (ryancarlson) [inline src/main/java/org/cru/godtools/domain/images/ImageService.java]: Good call.

**Comment by ryancarlson** [comment]
@LaelWatkins I believe I've addressed your feedback here, as well as patched the cause of that NPE that cropped up.  Can you take another look when you have some time?

**Comment by LaelWatkins** [comment]
really nice! 👍

**Comment by ryancarlson** [comment]
Thanks @LaelWatkins

**Pull request merged**